core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -618,4 +628,32 @@ public HttpResponse convertResponse(ServiceRequestContext ctx,
     private enum ResponseType {
         HTTP_RESPONSE, COMPLETION_STAGE, KOTLIN_COROUTINES, SCALA_FUTURE, OTHER_OBJECTS
     }
+
+    /**
+     * Converts {@link Method} to {@link MethodHandle}, optionally accepting {@code object} instance of the
+     * declaring class in case of non-static methods. Result {@link MethodHandle} must be assigned to
+     * a {@code final} field in order to enable Java compiler optimizations.
+     * @param method the {@link Method} to be converted to a {@link MethodHandle}
+     * @param object an instance of declaring class for non-static methods, or {@link null} for static methods
+     * @return a {@link MethodHandle} corresponding to the supplied {@link Method}
+     */
+    private static MethodHandle asMethodHandle(Method method, @Nullable Object object) {
```

Comment:
It seems like `object` is always not null.

Reply:
indeed, in this context it's not null. reserved it to Nullable in case of static methods

Reply:
This is a private method, so could we optimize the the signature for the current scope?
The caller method already makes sure the `object` is not null.